src/Compression/CompressionCodecEncrypted.h

```diff
@@ -79,16 +85,19 @@ namespace DB
 
     private:
         static std::string lastErrorString();
-        static std::string deriveKey(const std::string_view & master_key);
         static void encrypt(const std::string_view & plaintext, char * ciphertext_and_tag);
-        static void decrypt(const std::string_view & ciphertext_and_tag, char * plaintext);
+        static void decrypt(const std::string_view & ciphertext_and_tag, char * plaintext, UInt64 key_id=current_key_id);
+
+
+        static std::unordered_map<UInt64, String> keys_storage;
+        static UInt64 current_key_id;
```

Comment:
It's better to avoid static functions doing stuff with a bunch of static variables, because it may easily produce lots of troubles with synchronization. I suppose we can introduce a class-singleton instead, something like

```
class CompressionCodecEncrypted
{
    ...
    class Configuration
    {
    public:
        static Configuration & instance();
        
        void load(const Poco::Util::AbstractConfiguration & config, const String & config_prefix);
        String getKey(UInt64 id) const;
        String getCurrentKey() const;

    private:
        struct Params
        {
             std::unordered_map<UInt64, String> keys_storage;
             UInt64 current_key_id;
        };
        MultiVersion<Params> params;
    };
};
```

`MultiVersion` is useful here because it helps to synchronize updating `params` (we need synchronization here because `load()` and `getKey()` can be called from different threads at the same time).